rotkehlchen/tests/api/test_snapshots.py

```diff
@@ -0,0 +1,178 @@
+import csv
+import os
+import tempfile
+import zipfile
+from http import HTTPStatus
+from pathlib import Path
+
+import requests
+
+from rotkehlchen.data_handler import DBHandler
+from rotkehlchen.db.snapshots import BALANCES_FILENAME, LOCATION_DATA_FILENAME
+from rotkehlchen.tests.utils.api import api_url_for, assert_proper_response
+from rotkehlchen.user_messages import MessagesAggregator
+
+
+def _populate_db_with_balances(connection):
+    cursor = connection.cursor()
+    cursor.execute(
+        """
+        INSERT INTO "timed_balances" ("category", "time", "currency", "amount", "usd_value") VALUES
+        (?, ?, ?, ?, ?);
+        """, ('A', 1, 'ADA', '1.00', '178.44'),
+    )
+    cursor.execute(
+        """
+        INSERT INTO "timed_balances" ("category", "time", "currency", "amount", "usd_value") VALUES
+        (?, ?, ?, ?, ?);
+        """, ('A', 1, 'AVAX', '1.00', '87'),
+    )
+    connection.commit()
+
+
+def _populate_db_with_location_data(connection):
+    cursor = connection.cursor()
+    cursor.execute(
+        """
+        INSERT INTO "timed_location_data" ("time", "location", "usd_value") VALUES
+        (?, ?, ?);
+        """, (1, 'A', '100.00'),
+    )
+    cursor.execute(
+        """
+        INSERT INTO "timed_location_data" ("time", "location", "usd_value") VALUES
+        (?, ?, ?);
+        """, (1, 'B', '200.00'),
+    )
+    cursor.execute(
+        """
+        INSERT INTO "timed_location_data" ("time", "location", "usd_value") VALUES
+        (?, ?, ?);
+        """, (1, 'H', '50.00'),
+    )
+    connection.commit()
+
+
+def assert_csv_export_response(response, csv_dir, is_download=False):
+    if is_download:
+        assert response.status_code == HTTPStatus.OK
+    else:
+        assert_proper_response(response)
+        data = response.json()
+        assert data['message'] == ''
+        assert data['result'] is True
```

Comment:
We have a utility test function in tests/utils/api.py that does that all.

```suggestion
        assert_simple_ok_response(response)
```